src/MongoDB.Driver.Core/Core/Connections/IsMasterResult.cs

```diff
@@ -322,6 +322,23 @@ public ServerType ServerType
             }
         }
 
+        /// <summary>
+        /// Get the SpeculativeAuthenticate reply.
+        /// </summary>
+        /// <value>
+        /// Null if isMaster["ok"] != 1 or if the SpeculativeAuthenticate reply was not included in the isMaster response.
+        /// </value>
+        public BsonDocument SpeculativeAuthenticate
+        {
+            get
+            {
+                return _wrapped.TryGetValue("ok", out var ok) && ok.ToBoolean() &&
+                       _wrapped.TryGetValue("speculativeAuthenticate", out var speculativeAuthenticate)
+                    ? (BsonDocument) speculativeAuthenticate
```

Comment:
>it often results in extraneous changes

Just to note: I didn't see any wrong changes for now. And I think it's one of the best ways to avoid styling comments at least when it's possible.